src/bebone/core/types/matrix/matrix_4x4.h

```diff
@@ -0,0 +1,267 @@
+#ifndef _BEBONE_CORE_TYPES_MATRIX_4X4_TPP_
+#define _BEBONE_CORE_TYPES_MATRIX_4X4_TPP_
+
+#include "matrix.tpp"
+
+namespace bebone::core {
+    template<>
+    struct Matrix<f32, 4, 4> {
+    public:
+        f32 e[4][4];
+
+    public:
+        Matrix() = default;
+
+        Matrix(f32 n00, f32 n01, f32 n02, f32 n03,
+               f32 n10, f32 n11, f32 n12, f32 n13,
+               f32 n20, f32 n21, f32 n22, f32 n23,
+               f32 n30, f32 n31, f32 n32, f32 n33);
+
+        Matrix(const Vec4f& a, const Vec4f& b, const Vec4f& c, const Vec4f& d);
+
+        inline static Matrix<f32, 4, 4> splat(const f32& value);
+        inline static Matrix<f32, 4, 4> identity();
+
+        static Matrix<f32, 4, 4> get_rotation_x(const f32& angle);
+        static Matrix<f32, 4, 4> get_rotation_y(const f32& angle);
+        static Matrix<f32, 4, 4> get_rotation_z(const f32& angle);
+        static Matrix<f32, 4, 4> get_rotation_matrix(const f32& angle, Vec3f axis);
+        static Matrix<f32, 4, 4> get_reflection_matrix(const Vec3f& v);
+        static Matrix<f32, 4, 4> get_involution_matrix(const Vec3f& v);
+
+        static Matrix<f32, 4, 4> skew(f32 t, Vec3f a, Vec3f b);
+
+        static Matrix<f32, 4, 4> scale(const f32& s);
+        static Matrix<f32, 4, 4> scale(const f32& sx, const f32& sy, const f32& sz);
+        static Matrix<f32, 4, 4> scale(const Vec3f& s);
+        static Matrix<f32, 4, 4> scale(f32 s, Vec3f v);
+
+        static Matrix<f32, 4, 4> translation(const f32& tx, const f32& ty, const f32& tz);
+        static Matrix<f32, 4, 4> translation(const Vec3f& t);
+
+        static Matrix<f32, 4, 4> orthographic(const f32& right, const f32& left, const f32& bottom, const f32& top, const f32& near, const f32& far);
+        static Matrix<f32, 4, 4> perspective(f32 fov, const f32& aspect, const f32& near, const f32& far);
+        static Matrix<f32, 4, 4> view(const Vec3f& origin, const Vec3f& direction, const Vec3f& up);
```

Comment:
Actually, view matrix is a derivation from look_at function (look_at is more like a standard, cause for glm library), look_at is very similar to view, but instead of ``direction`` it accepts ``center``
Direction is computed using camera origin and look point(center). Usually we use look_at function then we need orbit camera, it allows to place camera at some place, and face it perfectrly toward some point

Implementation is very similar with view (this is from bepbep repo)
![image](https://github.com/user-attachments/assets/d3341e6c-ee0c-493f-b37b-04e6b3112bf4)

Reply:
Just add this look_at function, it will be useful in the future, no need to change view implementation

Reply:
Ok!